src/keygen.rs

```diff
@@ -75,23 +69,13 @@ impl<C: Ciphersuite> From<keygen_protocol::Error<C>> for Error {
 /// # Note
 /// - `ciphersuite`: The `ID` of the ciphersuite; oneof [`FROST-ED25519-SHA512-v1`, `FROST-secp256k1-SHA256-v1`].
 /// - `threshold`: The threshold of the keygen protocol should be less than the number of operators.
-#[sdk::job(
-    id = 0,
-    params(ciphersuite, threshold),
-    result(_),
-    event_listener(
-        listener = TangleEventListener::<FrostContext, JobCalled>,
-        pre_processor = services_pre_processor,
-        post_processor = services_post_processor,
-    )
-)]
 #[tracing::instrument(target = "gadget", skip(context), err)]
 pub async fn keygen(
-    ciphersuite: String,
-    threshold: u16,
-    context: FrostContext,
-) -> Result<Vec<u8>, Error> {
-    logging::info!("Running keygen job");
+    CallId(current_call_id): CallId,
+    Context(context): Context<FrostContext>,
+    TangleArgs2(ciphersuite, threshold): TangleArgs2<String, u16>,
```

Comment:
Weird name no?

Reply:
Ah, so we used to `TangleArg` which is fine for the most blueprints that only accept one arg, that arg could be a struct or a tuple. However if you want to have more arguments to your job, you have to use `TangleArgs2..16`. If you have a better names, please suggest.

Reply:
Is there no way to implement some clever thing for tuples?

Reply:
You can use `TangleArg<(T, T2, T3, ...)>` but you have to send your types as a tuples on-chain too, not as multiple args.